backend/mem/location.go

```diff
@@ -0,0 +1,202 @@
+package mem
+
+import (
+	"errors"
+	"path"
+	"regexp"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/c2fo/vfs/v5"
+	"github.com/c2fo/vfs/v5/utils"
+)
+
+//Location implements the vfs.Location interface specific to in-memory FileSystem.
+type Location struct {
+	exists     bool
+	firstTime  bool
+	name       string //the path that this location exists on
+	fileSystem *FileSystem
+	volume     string
+}
+
+//String implements io.Stringer by returning the location's URI as a string
+func (l *Location) String() string {
+
+	return l.URI()
+}
+
+/*
+List finds all of the files living at the current location and returns them in a slice of strings.
+If there are no files at location, then an empty slice will be returned
+*/
+
+func (l *Location) List() ([]string, error) {
+
+	str := l.Path()                         //full path of this location
+	mapRef := &l.fileSystem.fsMap           //setting mapRef to this value for code readability
+	if _, ok := (*mapRef)[l.Volume()]; ok { //are there paths on this volume?
+		list := (*mapRef)[l.Volume()].fileNamesHere(str) //getting a list of the file names on this location
+
+		return list, nil // "fileNamesHere" returns an empty list if no files were found
+	}
+	return make([]string, 0), nil //if the volume has nothing on it, return an empty list as well
+}
+
+/*
+ListByPrefix tags a prefix onto the current path and in a slice,
+ returns all file base names whose full paths contain that substring
+Returns empty slice if nothing found
+*/
+
+func (l *Location) ListByPrefix(prefix string) ([]string, error) {
+
+	list := make([]string, 0)
+	str := path.Join(l.Path(), prefix)
+	mapRef := &l.fileSystem.fsMap
```

Comment:
I see this patter a lot:

```
mapRef := &l.fileSystem.fsMap
(*mapRef)[l.volume]
```

Why not just get the map instead of a pointer to the map? It cleans up syntax a lot.


```
mapRef := l.fileSystem.fsMap
mapRef[l.volume]
```